pkg/apis/build/v1alpha1/buildrun_types.go

```diff
@@ -64,8 +64,50 @@ type BuildRunStatus struct {
 	// BuildSpec is the Build Spec of this BuildRun.
 	// +optional
 	BuildSpec *BuildSpec `json:"buildSpec,omitempty"`
+
+	//stages containes the details about the stage for each build with stage start time 
+	// completion and duration
+	// +optional
+	Stages []StageInfo `json:"stages,omitempty"`
+}
+
+// StageInfo contains details about a build stage.
+type StageInfo struct {
+	// name is a unique identifier for each stage of the builds
+	Name string `json:"name,omitempty"`
```

Comment:
If it is omitempty, we should have a `+optional` tag as well. The same goes for other endpoints.